### DISCUSSION TITLE
Question regarding noslip wall on EB

### DISCUSSION THREAD

**Discussion by Pandakii** [AMReX-Combustion/PeleC#626] [Q&A]
Hi，

I am trying to simulate the supersonic flow inside a cavity flame-holder using PeleC. The irregular lower wall  is built by function initialize_EB2(). In my test case, I use the command 'pelec.eb_noslip = 1 ' in the input file  to set the EB condition as noslip wall. However, the test result shows that there is no boundary layer on the EB. The EB condition seemingly is slip wall. I do not know why the command 'pelec.eb_noslip = 1 ' is useless. This is my input file [code-inp.txt](https://github.com/AMReX-Combustion/PeleC/files/11551665/code-inp.txt).

I would appreciate it if anyone could help me.
Thanks in advance.

![model](https://github.com/AMReX-Combustion/PeleC/assets/132762514/a217a3ec-5384-4bbc-88d4-d4445e2051cf)
![contours](https://github.com/AMReX-Combustion/PeleC/assets/132762514/a0d42aef-431b-4286-930b-0be744eee978)

**Comment by marchdf** [comment]
This is indeed strange and not what I would expect. Are you using your own InitEB.cpp? We have a bunch of test cases that should catch this kind of thing. Including: https://amrex-combustion.github.io/PeleC/ebverification/C10/README.html#eb-c10 (https://github.com/AMReX-Combustion/PeleC/tree/development/Exec/RegTests/EB-C10). Can you make sure there isn't anything different in the input files/code you are using?

**Reply by Pandakii** [reply]
Thanks very much for your reply. I modified the function initialize_EB2() in the InitEB.cpp to generate the EB. This is the InitEB.cpp in my case [code-initEB_cpp.txt](https://github.com/AMReX-Combustion/PeleC/files/11560404/code-initEB_cpp.txt)
.  I also checked my input file and compared it to the test case EB-C10. The diffuse_spec and diffuse_enth are shut down in the EB-C10  but they are switched on in my case. I'm not sure if this difference is responsible for this strange result.

**Reply by marchdf** [reply]
that shouldn't make a difference... Ok instead of modifying InitEB, I would recommend you add your geometry to Geometry.H/cpp. Lots of examples in there on how to do that.

**Reply by Pandakii** [reply]
Thanks for your comment. I try to use the extruded_triangles to generate a simple  EB wall instead of my own  InitEB.cpp to generate a EB wall. However, the issure also occurs. Do I need to upload other files?
 ![ET](https://github.com/AMReX-Combustion/PeleC/assets/132762514/b920ee87-35c4-4204-9994-a58c1926bc46)

**Reply by Pandakii** [reply]
Hi Marc, 

In my recent test, I  use the Constant transport model in my case and add the viscosity  parameter  in the prob.cpp. The viscous EB could bee seen. Does the noslip EB conditon only work under the Constant transport model?

**Reply by marchdf** [reply]
It should work for all transport models. I just tried the `ChannelFlow` test case (uses Simple and air). I modified it to not do velocity perturbations and threw an EB in there. It develops a gradient at the EB boundary so I think it's all fine there too... 

<img width="892" alt="Screen Shot 2023-06-05 at 09 01 35" src="https://github.com/AMReX-Combustion/PeleC/assets/15038415/9a344ef2-e669-4652-b61a-f1894d06ac16">

**Reply by Pandakii** [reply]
Thanks for your reply. I run the ChannelFlow test case and it works fine.

**Reply by Pandakii** [reply]
Hi Marc,

I modified the left boundary as the velocity inlet in the ChannelFlow case and only one side wall was set as  EB.  I found that the density and temperature gradients  were different on the EB and domain boundary.  It is  strange and I am puzzled by this difference. 
![DT](https://github.com/AMReX-Combustion/PeleC/assets/132762514/fceb9da1-322a-4fab-8b89-ae2fae46dc43)

**Reply by marchdf** [reply]
this is odd... but I did have a thought yesterday. Can you plot `viscosity`? you have to add that keyword to the list of output vars.

**Reply by Pandakii** [reply]
This is the viscosity contour. The EB has the lower viscosity.
![viscosity](https://github.com/AMReX-Combustion/PeleC/assets/132762514/009d5c60-16b9-494f-a504-efa83b2fceb3)

**Comment by hsitaram** [comment]
I wonder if this is a plotting issue? is this done using visit? I didnt see anything obvious in our implementation as our viscous test cases work fine.

**Reply by marchdf** [reply]
Yes I would further say that these are cell center values being plotted so the velocity won't be zero in that cell.

**Reply by Pandakii** [reply]
Yes, I use visit to plot the contour. I try to move away the outline but no obvious velocity gradient could be seen on the EB. So, I do not know why there is no boundary layer on the EB.
![no outline](https://github.com/AMReX-Combustion/PeleC/assets/132762514/ad2369a0-6fa1-4fce-9853-74dc57825556)

**Reply by hsitaram** [reply]
I also noticed that you have only 8 cells along the out of plane direction while setting lo and hi_bcs as walls. In this [paper](https://www.sciencedirect.com/science/article/pii/S0010218021002741) we had used periodic bcs in the out of plane direction. I am wondering if the out of plane under-resolution may be the issue. Would you be able to rerun with periodic bcs in the z direction?

**Reply by marchdf** [reply]
I just ran EB-C10 to make sure nothing broke without us noticing. I definitely see a gradient at the wall... Is the bottom of your cavity an EB wall or just the domain BC set to a wall? Are you seeing a gradient there (I can't tell from your picture)?

![Screen Shot 2023-05-25 at 09 51 22](https://github.com/AMReX-Combustion/PeleC/assets/15038415/c5f3afec-ef1f-4655-b962-88a9a471e1fa)

**Reply by Pandakii** [reply]
> I also noticed that you have only 8 cells along the out of plane direction while setting lo and hi_bcs as walls. In this [paper](https://www.sciencedirect.com/science/article/pii/S0010218021002741) we had used periodic bcs in the out of plane direction. I am wondering if the out of plane under-resolution may be the issue. Would you be able to rerun with periodic bcs in the z direction?
Thanks your comment.  Please forgive me for not emphasizing that my case is two-dimensional. Is the issue causd by the 2D EB? Is it necessary to test a 3D case to compare result?

**Reply by marchdf** [reply]
2D should work.. Here is C10 with 2D: 
![Screen Shot 2023-05-25 at 11 07 13](https://github.com/AMReX-Combustion/PeleC/assets/15038415/bc56a5cd-5061-4f41-8b79-448c1620ecd7)

**Reply by Pandakii** [reply]
> I just ran EB-C10 to make sure nothing broke without us noticing. I definitely see a gradient at the wall... Is the bottom of your cavity an EB wall or just the domain BC set to a wall? Are you seeing a gradient there (I can't tell from your picture)?
> 
> ![Screen Shot 2023-05-25 at 09 51 22](https://user-images.githubusercontent.com/15038415/240968096-c5f3afec-ef1f-4655-b962-88a9a471e1fa.png)

The cavity bottom wall is the domain BC set to a wall.  I checked my result and the gradient could form on the bottom wall. The issue only occurs on the EB.
